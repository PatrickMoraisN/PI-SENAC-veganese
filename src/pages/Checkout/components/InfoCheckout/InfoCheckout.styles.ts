import styled from 'styled-components'

export const InfoCheckoutContainer = styled.div`
  margin-top: 25px;
`

export const InfoWrapper = styled.div`
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: space-between;
  margin-top: 15px;
`

export const TotalText = styled.p`
  font-weight: 700;
  font-size: 20px;
  color: #403937;
`
