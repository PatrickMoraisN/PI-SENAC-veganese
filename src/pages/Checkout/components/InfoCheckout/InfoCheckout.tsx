import { Localization } from '@/components/Localization/Localization'
import React, { useEffect, useState } from 'react'
import * as S from './InfoCheckout.styles'

interface InfoCheckoutProps {
  totalItensValue: number
  setShippingCalculated: (value: boolean) => void
  setError: (value: null) => void
  error: string | null
}

export function InfoCheckout({
  totalItensValue,
  setShippingCalculated,
  error,
  setError,
}: InfoCheckoutProps) {
  const [shipping, setShipping] = useState(0)

  const total = shipping + totalItensValue

  useEffect(() => {
    if (shipping) {
      setError(null)
    }
  }, [shipping])

  return (
    <S.InfoCheckoutContainer>
      <S.InfoWrapper>
        <p>Total de itens</p>
        <p>R$ {totalItensValue.toFixed(2)}</p>
      </S.InfoWrapper>
      <S.InfoWrapper>
        <Localization
          setShipping={setShipping}
          error={error}
          setShippingCalculated={setShippingCalculated}
        />
      </S.InfoWrapper>
      <S.InfoWrapper>
        {!!shipping && (
          <>
            <S.TotalText>Total</S.TotalText>
            <S.TotalText>R$ {total.toFixed(2)}</S.TotalText>
          </>
        )}
      </S.InfoWrapper>
    </S.InfoCheckoutContainer>
  )
}
