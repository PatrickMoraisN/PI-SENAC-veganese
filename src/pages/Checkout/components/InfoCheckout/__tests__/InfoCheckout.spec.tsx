import { render } from '@testing-library/react'
import { InfoCheckout } from '../InfoCheckout'

const setShippingMock = jest.fn()
const setErrorMock = jest.fn()
const totalItensValueMock = 29.943

const renderInfoCheckout = () =>
  render(
    <InfoCheckout
      totalItensValue={totalItensValueMock}
      setShippingCalculated={setShippingMock}
      error={null}
      setError={setErrorMock}
    />,
  )

describe('InfoCheckout', () => {
  it('should render correcly', () => {
    const { container } = renderInfoCheckout()

    expect(container).toMatchSnapshot()
  })
})
