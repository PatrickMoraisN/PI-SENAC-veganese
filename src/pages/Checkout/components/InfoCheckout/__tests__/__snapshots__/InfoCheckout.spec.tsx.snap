// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCheckout should render correcly 1`] = `
<div>
  <div
    class="sc-gswNZR lbiwJV"
  >
    <div
      class="sc-dkrFOg hQptSf"
    >
      <p>
        Total de itens
      </p>
      <p>
        R$ 
        29.94
      </p>
    </div>
    <div
      class="sc-dkrFOg hQptSf"
    >
      <button
        class="sc-bcXHqe kyvJqT"
      >
        Calcular Frete
      </button>
    </div>
    <div
      class="sc-dkrFOg hQptSf"
    />
  </div>
</div>
`;
